.github/workflows/continuous-integration-workflow.yml

```diff
@@ -0,0 +1,75 @@
+name: CI
+on:
+  push:
+    branches:
+      - master
+      - develop
+  pull_request:
+    branches:
+      - master
+      - develop
+  schedule:
+    - cron:  '0 5 * * FRI'
+
+jobs:
+  CI:
+    continue-on-error: ${{ matrix.distro == 'ubuntu:devel' || matrix.distro == 'opensuse:latest' || matrix.distro == 'fedora:rawhide' || matrix.continue-on-error == true }}
+    strategy:
+      matrix:
+        distro: ['fedora:latest', 'fedora:rawhide', 'opensuse:latest', 'ubuntu:latest', 'ubuntu:devel', 'ubuntu:rolling']
```

Comment:
This looks a bit excessive for CI. If at all, I would test multiple distros only on merge commits (and not on every commit in a pull request) as a regression test.

Reply:
The builds are relatively quick, so I didn't do a pre-selection, but we can do whatever. To build on different distros for PRs vs. pushes on master, we would need to split this into 2 files. Fine with me, just saying.

Reply:
56 builds seem not overburdensome, but it is your call.